mpazaryna/aiforge
# feat(logging): enhance logging and add API interaction tracking

# feat(logging): enhance logging and add API interaction tracking

- Add detailed logging for risk assessment model execution
- Implement logging for pytest test cases
- Include JSON output logging for risk assessment results
- Add debug logging for LangChainService initialization and API calls
- Implement logging for HTTP requests and responses
- Add asyncio debug logging
- Include OpenAI API interaction logging

test: add pytest logging tests

- Implement test cases for various log levels (DEBUG, INFO, WARNING, ERROR)

chore(deps): update OpenAI client and add HTTP client logging

- Update OpenAI client to use new base client
- Add detailed HTTP client logging for API interactions

refactor(api): improve error handling and response generation

- Enhance error handling for LangChainService initialization
- Refactor response generation process

ci: add health check and alert system

- Implement health check endpoints
- Add alert check system

This commit significantly improves the logging capabilities across the
application, enhances testing, and provides better visibility into API
interactions and system health.
